File: python_code/helpers/numbers.py
def pow_and_fix(f_value, decimals, offset=1):
    s_value = str(f_value)
    precision = len(s_value) - (s_value.find('.') + 1)
    stage_a = round(f_value * pow(10, precision))
    stage_b = pow(10, decimals - precision)
    stage_c = (stage_a * stage_b)
    result = round(stage_c, offset)
    print('f_value', f_value, 'decimals', decimals, 'offset', offset,
          's_value', s_value, 'precision', precision,
          'stage_a', stage_a, 'stage_b', stage_b, 'stage_c', stage_c,
          'result', result)
    return result

File: node_code/src/helpers/numbers.ts
export function powAndFix(fValue: number, decimals: number, offset: number = 1): number {
  const sValue: string = fValue.toString();
  const precision: number = sValue.length - ((sValue.indexOf('.') || sValue.length) + 1);
  const stageA: number = Math.round(fValue * Math.pow(10, precision));
  const stageB: number = Math.pow(10, decimals - precision);
  const stageC: number = (stageA * stageB) * Math.pow(10, offset);
  const result: number = Math.round(stageC) / Math.pow(10, offset);
  console.log('fValue', fValue, 'decimals', decimals, 'offset', offset,
    'sValue', sValue, 'precision', precision,
    'stageA', stageA, 'stageB', stageB, 'stageC', stageC,
    'result', result);
  return result;
}
